File: Exercise/Medium/09. Pangram Checker/Code/Pangram Checker.cpp
#include <iostream>
#include <string>
#include <cctype>

using namespace std;

bool isPangram(const string& str) {
    bool alphabet[26] = {false};
    int index;

    for (char ch : str) {
        if (isalpha(ch)) {
            index = tolower(ch) - 'a';
            alphabet[index] = true;
        }
    }

    for (bool present : alphabet) {
        if (!present) {
            return false; 
        }
    }
    return true; 
}

int main() {
    string str;
    getline(cin, str);

    if (isPangram(str)) {
        cout << "True" << endl;
    } else {
        cout << "False" << endl;
    }

    return 0;
}

File: Exercise/Medium/09. Pangram Checker/Code/Pangram Checker.py
def is_pangram(input_str):
    alphabet = [False] * 26

    for ch in input_str:
        if ch.isalpha():
            index = ord(ch.lower()) - ord('a')
            alphabet[index] = True

    for present in alphabet:
        if not present:
            return False
    
    return True

def main():
    input_str = input()

    if is_pangram(input_str):
        print("True")
    else:
        print("False")

if __name__ == "__main__":
    main()